src/blob/index/core.rs

```diff
@@ -54,21 +53,82 @@ where
 }
 
 #[derive(Debug, Default)] // Default can be used to initialize structure with 0
-pub(crate) struct MemoryAttrs {
-    pub(crate) key_size: usize,
-    pub(crate) btree_entry_size: usize,
-    pub(crate) records_allocated: usize,
+struct MemoryAttrs<K> {
+    records_count: usize,
+    records_allocated: usize,
+    marker: PhantomData<K>,
+}
+
+impl<K> MemoryAttrs<K>
+where
+    for<'a> K: Key<'a>,
+{
+    const fn key_size(&self) -> usize {
+        K::LEN as usize
+    }
+
+    const fn bptree_entry_size(&self) -> usize {
```

Comment:
According to the documentation `const fn` works as a regular function, but additionally they can be used to initialize a constant. That means, that it is up to the optimizer to fold all this function into a single constant. I've checked, that on debug level Rust compiler emits a full calculation logic. Thus I suggest to declare a regular constant here instead of `const fn`:

```
const KEY_SIZE: usize = K::LEN as usize;
const BPTREE_ENTRY_SIZE: usize = size_of::<Vec<u8>>() + KEY_SIZE + size_of::<Vec<RecordHeader>>();
const RECORD_HEADER_SIZE: usize = size_of::<RecordHeader>() + KEY_SIZE;
```

You can keep these functions for simplicity, but they should return a declared constants and be attributed with `#[inline]`

Reply:
I don't think functions are needed here, replaced with `const`s